source/agora/node/Registry.d

```diff
@@ -752,14 +766,23 @@ private struct ZoneData
         this.config = config;
         this.root = root;
 
-        static string serverType (bool auth, bool primary)
+        if (this.config.authoritative)
         {
-            return primary ? "primary" : (auth ? "secondary" : "caching");
+            if (this.config.soa.email.set)
+                this.type = ZoneType.primary;
+        }
+
+        static string serverType (ZoneType zone_type)
+        {
+            switch (zone_type)
+            {
+                case ZoneType.primary: return "primary (authoritative)";
+                default: return "Unknown";
```

Comment:
I know this is fixed in a later commit, but we should not return "Unknown" here.